dev/com.ibm.ws.microprofile.config_repeat_tests/src/com/ibm/ws/microprofile/config/fat/repeat/RepeatConfigActions.java

```diff
@@ -10,121 +10,53 @@
  *******************************************************************************/
 package com.ibm.ws.microprofile.config.fat.repeat;
 
-import componenttest.custom.junit.runner.Mode.TestMode;
-import componenttest.custom.junit.runner.TestModeFilter;
+import componenttest.rules.repeater.FeatureReplacementAction;
 import componenttest.rules.repeater.RepeatTests;
 
 /**
  *
  */
 public class RepeatConfigActions {
 
-    /**
-     * Get the RepeatTests actions for Config 1.1 tests. This is the same as for Config 1.2 tests but
-     * if the test mode is FULL then it also adds Config 1.1 + EE7 and Config 1.1 + EE8.
-     *
-     * @param server The name of the test server
-     * @return The RepeatTests for Config 1.1 tests
-     */
-    public static RepeatTests repeatConfig11(String server) {
-        RepeatTests r = repeatConfig12(server);
-        if (TestModeFilter.shouldRun(TestMode.FULL)) {
-            r = r.andWith(new RepeatConfig11EE7(server));
-            r = r.andWith(new RepeatConfig11EE8(server));
+    public enum Version {
+        CONFIG11_EE7, CONFIG11_EE8, CONFIG12_EE7, CONFIG12_EE8, CONFIG13_EE7, CONFIG13_EE8, CONFIG14_EE7, CONFIG14_EE8, CONFIG20_EE8, LATEST
+    };
+
+    public static FeatureReplacementAction getAction(Version version, String server) {
+        switch (version) {
+            case CONFIG11_EE7:
+                return new RepeatConfig11EE7(server);
+            case CONFIG11_EE8:
+                return new RepeatConfig11EE8(server);
+            case CONFIG12_EE7:
+                return new RepeatConfig12EE7(server);
+            case CONFIG12_EE8:
+                return new RepeatConfig12EE8(server);
+            case CONFIG13_EE7:
+                return new RepeatConfig13EE7(server);
+            case CONFIG13_EE8:
+                return new RepeatConfig13EE8(server);
+            case CONFIG14_EE7:
+                return new RepeatConfig14EE7(server);
+            case CONFIG14_EE8:
```

Comment:
Is this correct? I know `LATEST` covers this option, but should `CONFIG14_EE8` be defined if it does nothing?

Reply:
Okay, I see `CONFIG14_EE8 ` will just do whatever `LATEST` does, but I think `CONFIG14_EE8` should have `return new RepeatConfig14EE8(server);` for readability and maintainability